src/components/orders/OrderNotFound/index.tsx

```diff
@@ -69,21 +70,48 @@ const Support = styled.a`
     text-decoration: none;
   }
 `
+interface LocationState {
+  referrer: string
+}
+export const OrderAddressNotFound: React.FC = (): JSX.Element => {
+  const { searchString } = useParams<{ searchString: string }>()
+  const location = useLocation<LocationState>()
+  const history = useHistory()
+  const { referrer } = location.state || { referrer: null }
+  const wasRedirected = referrer ? true : false
+
+  // used after refresh by remove referrer state if was redirected
+  useEffect(() => {
+    window.addEventListener('beforeunload', () => {
+      history.replace(location.pathname, null)
+    })
 
-export const OrderNotFound: React.FC = () => {
-  const { orderId } = useParams<{ orderId: string }>()
+    return (): void => {
+      window.removeEventListener('beforeunload', () => {
+        history.replace(location.pathname, null)
+      })
+    }
+  }, [history, location.pathname])
 
   return (
     <>
-      <Title>Order not found</Title>
+      <Title>Order or Address not found</Title>
       <Content>
-        <p>Sorry, no matches found for:</p>
-        <p>
-          <strong>&quot;{orderId}&quot;</strong>
-        </p>
+        {searchString ? (
+          <>
+            <p>Sorry, no matches found for:</p>
+            <p>
+              <strong>&quot;{searchString}&quot;</strong>
+            </p>
+          </>
+        ) : (
+          <>
+            <p>The search cannot be empty</p>
+          </>
```

Comment:
Nitpick: since you are returning a single element you don't need the fragment wrapping

```suggestion
          <p>The search cannot be empty</p>
```